solutions/usaco-1061.mdx

```diff
@@ -24,6 +24,89 @@ Now we can sort cows by their static coordinates, which ensures that earlier col
 
 **Time Complexity:** $\mathcal{O}(N^2)$
 <LanguageSection>
+<CPPSection>
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+int main() {
+	int N; cin >> N;
+	vector<array<int, 3>> Ncows, Ecows;
+
+	for (int i = 0; i < N; i++) {
+		string dir;
+		int x, y;
+		cin >> dir >> x >> y;
+
+		// we need to add i as well because we need to output them in order
+		if (dir == "N") {
+			Ncows.push_back({x, y, i});
+		}
+		else {
+			Ecows.push_back({x, y, i});
+		}
+	}
+
+	// sort north cows by x value
+	sort(Ncows.begin(), Ncows.end());
+
+	// sort east cows by y value
+	sort(Ecows.begin(), Ecows.end(), [](auto &left, auto &right) {
+		return left[1] < right[1];
+	});
+
+	// we sort them by x/y value because we know that, for a single cow, 
+	// it reaches a lower x/y value and then a higher one. 
+	// Thus, it is more we would not miss any earlier collisions
+	// if we iterate from lower to a higher value. 
+
+	// stores the amount of grass it eats
+	int stop[N];
+	for (int i = 0; i < N; i++) {
+		stop[i] = -1;
+  	}
+
+
+	for (auto Ncow : Ncows) {
+    		for (auto Ecow : Ecows) {
```

Comment:
`n_cos` and `e_cow`
also i'd like if you didn't use auto

Reply:
auto is fine